From c9adc4d4f76cd52e82842b1df8afc79b99f352d6 Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Sun, 23 Nov 2025 10:06:33 +0100
Subject: [PATCH 2/2] linker: Set LD_SHIM_LIBS via soong config
 shim,ld_shim_libs

Eg.: $(call soong_config_set,shim,ld_shim_libs,/vendor/lib64/libexynoscamera.so|/vendor/lib64/libexynoscamera_shim.so)

Change-Id: If0bc1284237df44b0c54cca45286c77b771cb35c
---
 linker/Android.bp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/linker/Android.bp b/linker/Android.bp
index 88fc09c90..5c028439c 100644
--- a/linker/Android.bp
+++ b/linker/Android.bp
@@ -90,6 +90,10 @@ cc_defaults {
 
     cflags: linker_common_flags,
     asflags: linker_common_flags,
+    cppflags: select(soong_config_variable("shim", "ld_shim_libs"), {
+                any @ flag_val: ["-DLD_SHIM_LIBS=\"" + flag_val + "\"", "-Wold-style-cast"],
+                default: ["-Wold-style-cast"],
+            }),
 
     product_variables: {
         debuggable: {
@@ -97,8 +101,6 @@ cc_defaults {
         },
     },
 
-    cppflags: ["-Wold-style-cast"],
-
     static_libs: [
         "libziparchive",
         "libbase",
-- 
2.51.0

